### PR TITLE
Checks if the environment.yml file is empty and if it contains a dict…

### LINKED ISSUE #304: Build failing on repo with Julia 0.6, NoneType object has no attribute get
via https://github.com/jupyterhub/binder/issues/73

> I am trying to build the following repo:
>
> https://mybinder.org/v2/gh/tpoisot/ScientificComputingForTheRestOfUs/master
>
> It was working up until last week-ish.
>
> When I try now, the build fails, and the only message I get is:
>
> ~~~
> Error during build: Build failing on repo with Julia 0.6, NoneType object has no attribute get
> ~~~
>
> The notebooks run on multiple machines locally.

### PR DESCRIPTION
**Pull request by GladysNalvarte**
…ionary.

If the environment.yml is empty, an empty dictionary is instantiated.
This is going to raise a TypeError if the environment.yml file contains a list
or other data structure different than a dictionary.

closes #304

### CODE REVIEW

File: tests/test_env_yml.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
+"""
+Test if the environment.yml is empty or it constains other data structure than a dictionary
+"""
+import os
+import sys
+import pytest
+from repo2docker import buildpacks
+
+#@pytest.mark.xfail(reason="Not yet implemented")
```
Comment (minrk) [inline tests/test_env_yml.py]: Can we remove this commented line?

File: tests/test_env_yml.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+def test_empty_env_yml(tmpdir):
+    tmpdir.chdir()
+    p = tmpdir.join("environment.yml")
```
Comment (minrk) [inline tests/test_env_yml.py]: I think we want a `p.write("")` to ensure there is an empty file here.

Comment (minrk) [reply tests/test_env_yml.py]: I think this one is also supposed to *not* raise.

File: tests/test_env_yml.py [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    bp = buildpacks.CondaBuildPack()
+    with pytest.raises(Exception):
+        py_ver = bp.python_version()
+
+def test_no_dict_env_yml(tmpdir):
+    tmpdir.chdir()
+    q = tmpdir.join("environment.yml")
+    q.write("list/n ,string")
+    bq = buildpacks.CondaBuildPack()
+    with pytest.raises(Exception):
```
Comment (minrk) [inline tests/test_env_yml.py]: Can this Exception be the specific `TypeError` that we are raising? To be sure that it's what we expect to happen?

File: tests/test_env_yml.py [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+import pytest
+from repo2docker import buildpacks
+
+#@pytest.mark.xfail(reason="Not yet implemented")
+def test_empty_env_yml(tmpdir):
+    tmpdir.chdir()
+    p = tmpdir.join("environment.yml")
+    bp = buildpacks.CondaBuildPack()
+    with pytest.raises(Exception):
+        py_ver = bp.python_version()
```
Comment (minrk) [inline tests/test_env_yml.py]: python_version is a `@property`, not a method, so this should be `py_ver = bp.python_version`

File: repo2docker/buildpacks/conda/__init__.py [hunk 5]
```diff
@@ -136,6 +136,12 @@ def python_version(self):
             py_version = None
             with open(environment_yml) as f:
                 env = YAML().load(f)
+                # check if the env file is empty, if so instantiate an empty dictionary.
+                if env is None:
+                    env = {}
+                # check if the env file has a dictionary not a list or other data structure.
+                if not isinstance(env, dict):
+                    raise TypeError("environment.yml should contain a dictionary. Got %r" % env)
```
Comment (choldgraf) [inline repo2docker/buildpacks/conda/__init__.py]: should this be `type(env)` at the end?

**Comment by choldgraf** [comment]
this LGTM and travis is happy, so I'm gonna merge! Thanks @GladysNalvarte !!!

**Pull request merged**